Importing toplevel packages conflicts with [Imports] sorting

## Configuration (required) ##
```
version = 3.5.7
runner.dialect = scala3
maxColumn = 100
rewrite.rules = [Imports]
rewrite.imports.sort = scalastyle
```

## Command-line parameters (required) ##

`scalafmt File.scala` (no other command line parameters)

## Steps

Given code like this:
```scala
import java as j
```

## Problem

Scalafmt errors out :
```
.../File.scala:1: error: [dialect scala3] identifier expected but . found
import .java as j
```

## Workaround

```scala
import __root__.java as j
```

## Notes
Probably related to https://github.com/lampepfl/dotty/pull/15321

--- a/scalafmt-core/shared/src/main/scala/org/scalafmt/rewrite/Imports.scala
+++ b/scalafmt-core/shared/src/main/scala/org/scalafmt/rewrite/Imports.scala
@@ -118,7 +118,7 @@ object Imports extends RewriteFactory {
         selectors: Selectors,
         owner: Importer
     ): Unit = {
-      val stat = s"$kw $ref.${selectors.pretty}"
+      val stat = s"$kw $ref${selectors.pretty}"
       if (stats.add(stat))
         buffer += GroupingEntry(stat, ref, selectors, owner)
     }
@@ -458,8 +458,8 @@ object Imports extends RewriteFactory {
       stats.foreach { x =>
         val kw = x.tokens.head.toString
         x.importers.foreach { importer =>
-          val ref = importer.ref.toString
-          val grp = groups(settings.group(ref + "."))
+          val ref = getRef(importer)
+          val grp = groups(settings.group(ref))
           addClauseToGroup(grp, kw, ref, importer)
         }
       }
@@ -644,9 +644,8 @@ object Imports extends RewriteFactory {
       stats.flatten.foreach { t =>
         val patchBuilder = Seq.newBuilder[TokenPatch]
         t.importers.foreach { importer =>
-          val ref = importer.ref.toString
           val selectors = getSelectors(importer.importees, false).pretty
-          val replacement = s"$ref.$selectors"
+          val replacement = getRef(importer) + selectors
           val tokens: Iterator[Token] = importer.tokens.iterator
           // replace the first token
           patchBuilder += TokenPatch.Replace(tokens.next(), replacement)
@@ -656,6 +655,11 @@ object Imports extends RewriteFactory {
         ctx.addPatchSet(patchBuilder.result(): _*)
       }
     }
+  }
+
+  private def getRef(importer: Importer): String = {
+    val ref = importer.ref.toString()
+    if (ref.isEmpty) "" else ref + "."
   }
 
 }

--- a/scalafmt-tests/src/test/resources/rewrite/Imports.source
+++ b/scalafmt-tests/src/test/resources/rewrite/Imports.source
@@ -905,3 +905,29 @@ import a.{x, given, given Z}
 import a.{x, y => z, given}
 import a.{*, given}
 import a.{x, given Z, given}
+<<< #3240 expand
+rewrite.imports.sort = ascii
+rewrite.imports.expand = true
+===
+import java as j
+>>>
+test does not parse
+import .java as j
+       ^
+<<< #3240 groups
+rewrite.imports.sort = ascii
+rewrite.imports.groups = [[".*"]]
+===
+import java as j
+>>>
+test does not parse
+import .java as j
+       ^
+<<< #3240 no expand, no groups
+rewrite.imports.sort = ascii
+===
+import java as j
+>>>
+test does not parse
+import .java as j
+       ^

--- a/scalafmt-tests/src/test/resources/rewrite/Imports.source
+++ b/scalafmt-tests/src/test/resources/rewrite/Imports.source
@@ -911,23 +911,17 @@ rewrite.imports.expand = true
 ===
 import java as j
 >>>
-test does not parse
-import .java as j
-       ^
+import java as j
 <<< #3240 groups
 rewrite.imports.sort = ascii
 rewrite.imports.groups = [[".*"]]
 ===
 import java as j
 >>>
-test does not parse
-import .java as j
-       ^
+import java as j
 <<< #3240 no expand, no groups
 rewrite.imports.sort = ascii
 ===
 import java as j
 >>>
-test does not parse
-import .java as j
-       ^
+import java as j

